src/Components/Authorization/AuthForm.tsx

```diff
@@ -0,0 +1,109 @@
+import React from "react"
+
+import ForgotPasswordForm from "./ForgotPasswordForm"
+import LoginForm from "./LoginForm"
+import RegisterForm from "./RegisterForm"
+import { FormComponentType, FormFields, Mode } from "./Types"
+
+type Validator = (val: string) => string
+
+interface Props extends React.HTMLProps<HTMLDivElement> {
+  mode: Mode
+  signupIntent?: string
+  destination?: string
+  formFields?: FormFields
+}
+
+interface State extends React.HTMLProps<HTMLDivElement> {
+  mode?: Mode
+  formFields: FormFields
+}
+
+class AuthForm extends React.Component<Props, State> {
+  static defaultProps: Partial<Props> = {
+    signupIntent: "sign_up",
+    destination: "/personalize", // ?
+    formFields: {
+      email: { value: "", error: "" },
+      name: { value: "", error: "" },
+      password: { value: "", error: "" },
+    },
+  }
+
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      mode: this.props.mode,
+      formFields: this.props.formFields,
+    }
+    this.handleUpdateInput = this.handleUpdateInput.bind(this)
+    this.handleChangeMode = this.handleChangeMode.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
```

Comment:
Can use => instead of binding in this way

```
handleUpdateInput = (event) => { ... } 
```

Fat arrow binds to the outer scope (`this`)

Reply:
Oh woops- I was trying to find the point in react history where people started using that style, and in the process was writing it both ways. I guess I kept it.

Reply:
Shout out to public class fields syntax.

Reply:
I get a little confused about this too @erikdstock the [react docs](https://reactjs.org/docs/handling-events.html) for handling events explains pretty well how and why you can use this syntax.

Reply:
(Here where the plugin is located: https://github.com/artsy/reaction/blob/master/.babelrc#L13)